src/huggingface_hub/hf_api.py

```diff
@@ -3574,6 +3574,33 @@ def update_repo_visibility(
         hf_raise_for_status(r)
         return r.json()
 
+    @validate_hf_hub_args
+    def update_repo_settings(
+        self,
+        repo_id: str,
+        *,
+        gated: Literal["auto", "manual", False] = False,
+        token: Union[str, bool, None] = None,
+        repo_type: Optional[str] = None,
+    ) -> Dict[str, Union[str, bool]]:
```

Comment:
```suggestion
    ) -> None:
```


@WizKnight I think your issue comes from the fact that the method expects a return value here. Since we're not returning anything, better to annotate with ` -> None`